src/services/services.ts

```diff
@@ -2034,7 +2064,13 @@ namespace ts {
         sourceMapText?: string;
     }
 
-
+    /**
+     * Matches a triple slash reference directive with an incomplete string literal for its path. Used
+     * to determine if the caret is currently within the string literal and capture the literal fragment
+     * for completions.
+     * For example, this matches /// <reference path="fragment
+     */
+    const tripleSlashDirectiveFragmentRegex = /^(\/\/\/\s*<reference\s+(path|types)\s*=\s*(?:'|"))([^'"]*)$/;

```

Comment:
The regex is actually only executed on the fragment of text leading up to the current caret position, so I believe matching the end of text is the correct behavior. I'll try and clarify that in the comment (I had the same thought earlier when I was reviewing my own code). Good catch on the `'` character, I'll make the change!